daml-lf/tests/MultiKeys.daml

```diff
@@ -130,7 +130,9 @@ template KeyOperations
          (keyCid, _) <- fetchByKey @Keyed p
          assert (keyCid `elem` [cid1, cid2])
          archive (fromSome (find (/= keyCid) [cid1, cid2]))
-         None <- lookupByKey @Keyed p
+         -- archive can only influence result for a contract fetched by key before.
```

Comment:
@andreaslochbihler-da no need to review the rest of this PR (of course you’re still welcome to), just an fyi that this changes archive semantics as you suggested which does imply a breaking change in multi-key semantics.

Reply:
Thanks for the heads-up.

@oliverse-da FYI: It might be that we need to adjust the `ContractKeysutra` test case when this change lands in Canton, but I don't expect any changes there.

Reply:
Thanks for the heads up.